src/ast/codegen_llvm.cpp

```diff
@@ -1300,38 +1300,54 @@ void CodegenLLVM::visit(Probe &probe)
 
     for (auto &attach_point : *probe.attach_points) {
       current_attach_point_ = attach_point;
-      std::set<std::string> matches;
+      std::unique_ptr<std::istream> symbol_stream;
+      std::string prefix, func;
       switch (probetype(attach_point->provider))
       {
         case ProbeType::kprobe:
         case ProbeType::kretprobe:
-          matches = bpftrace_.find_wildcard_matches(attach_point->target,
-                                                    attach_point->func,
-                                                    "/sys/kernel/debug/tracing/available_filter_functions");
+        {
+          symbol_stream = bpftrace_.get_symbols_from_file(
+              "/sys/kernel/debug/tracing/available_filter_functions");
+          prefix = "";
+          func = attach_point->func;
           break;
+        }
         case ProbeType::uprobe:
         case ProbeType::uretprobe:
         {
-          auto symbol_stream = std::istringstream(bpftrace_.extract_func_symbols_from_path(attach_point->target));
-          matches = bpftrace_.find_wildcard_matches("", attach_point->func, symbol_stream);
+          symbol_stream = std::make_unique<std::istringstream>(
+              bpftrace_.extract_func_symbols_from_path(attach_point->target));
+          prefix = "";
+          func = attach_point->func;
           break;
         }
         case ProbeType::tracepoint:
-          matches = bpftrace_.find_wildcard_matches(attach_point->target,
-                                                    attach_point->func,
-                                                    "/sys/kernel/debug/tracing/available_events");
+        {
+          symbol_stream = bpftrace_.get_symbols_from_file(
+              "/sys/kernel/debug/tracing/available_events");
+          prefix = attach_point->target;
+          func = attach_point->func;
           break;
+        }
         case ProbeType::usdt:
         {
-          auto usdt_symbol_stream = USDTHelper::probe_stream(bpftrace_.pid_, attach_point->target);
-          matches = bpftrace_.find_usdt_wildcard_matches(attach_point->ns, attach_point->func, usdt_symbol_stream);
+          symbol_stream = bpftrace_.get_symbols_from_usdt(
+              bpftrace_.pid_, attach_point->target);
+          prefix = "";
+          func = attach_point->ns + ":" + attach_point->func;
```

Comment:
Missed this yesterday - I'll do a bit more refactoring so that code doesn't need to be duplicated here